### PR TITLE
Changes to bootstrap contributing page

### PR DESCRIPTION
**Pull request by bulislaw**
Some changes to bootstrap contributing page.

@AnotherButler 
@sg-

### CODE REVIEW

File: docs/reference/contributing/target/bootstrap.md [hunk 1]
```diff
@@ -282,15 +287,15 @@ SECTIONS
...
-- Add an entry for your device in `mbed-os\targets\targets.json`
 
-#### Verification and next steps
+### Entry points
 
-Now verify that your target compiles by using `mbed compile -m MCU_NAME -t <toolchain>`.
+Except the reset vector, which is the standard entry point for Cortex-M cores, Mbed OS provides `mbed_sdk_init`, which the target can overload to perform higher level initialization. This function will be called later in the bootstrap process, after the basic initialization is done but before RTOS starts and before the `main` function is called.
 
-The next step to port a target is to enable the test harness dependencies. To run the test suite, your target must support GPIO, microsecond ticker and serial.
+Mbed OS provides another entry point that will be executed before `main` called `mbed_main`. This function is reserved for application use, and the target code should not define it.
```
Comment (AnotherButler) [inline docs/reference/contributing/target/bootstrap.md]: Query: Who or what executes the entry point before `main`? Also, who or what reserves the function for application use?

File: docs/reference/contributing/target/bootstrap.md [hunk 2]
```diff
@@ -282,15 +287,15 @@ SECTIONS
...
 - Define the initial stack pointer, `INITIAL_SP`, in `mbed_rtx.h`. This file is typically in `mbed-os\targets\TARGET_VENDOR\mbed_rtx.h`.
-- Add an entry for your device in `mbed-os\targets\targets.json`
 
-#### Verification and next steps
+### Entry points
 
-Now verify that your target compiles by using `mbed compile -m MCU_NAME -t <toolchain>`.
+Except the reset vector, which is the standard entry point for Cortex-M cores, Mbed OS provides `mbed_sdk_init`, which the target can overload to perform higher level initialization. This function will be called later in the bootstrap process, after the basic initialization is done but before RTOS starts and before the `main` function is called.
```
Comment (AnotherButler) [inline docs/reference/contributing/target/bootstrap.md]: Query: Who or what calls the function later in the bootstrap process? Also, who or what calls the `main` function?

Comment (bulislaw) [reply docs/reference/contributing/target/bootstrap.md]: Mbed OS internals.

File: docs/reference/contributing/target/bootstrap.md (hunk 1 above)
Comment (bulislaw) [reply docs/reference/contributing/target/bootstrap.md]: main is called by Mbed OS internals. As to who reserves the function it's 'us' as in we provide two functions `mbed_main` and `mbed_sdk_init` and we want one of them be available for platform code and second for the application code.

**Pull request merged**